test/libsolidity/smtCheckerTests/operators/conditional_assignment_statevar_1.sol

```diff
@@ -14,7 +14,5 @@ contract C {
 // ====
 // SMTEngine: all
 // ----
-// Warning 4984: (134-140): CHC: Overflow (resulting value larger than 2**256 - 1) might happen here.
```

Comment:
I got a bit confused by this at first, but do you agree that the `assert` is what prevents overflow from happening? Without the assert, `++a` could increment `a` until `2**256-1`, which would cause both `a + 10` and `++a` to overflow. But since `c >= a` must be true, it would already revert at `2**256-11`.

Reply:
I think the `assert` does not affect the overflow targets here. The overflow cannot happen in `a+10`, because that means `b` must be set to `true` in the constructor, but then the value of `a` never changes.
If `b` is set to `false`, then the overflow can happen, but it requires `2**256` calls to `f`. I guess that's why CHC runs into the time limit before finding the counterexample :)

Reply:
Yea, true. Even if we require the assertion to be true to continue with the execution path and `b = false`, even if `++a` overflows, `c` is still assigned `a` so the assertion should hold.